staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -375,11 +433,13 @@ func (c *Cacher) Watch(ctx context.Context, key string, resourceVersion string,
 		chanSize = 1000
 	}
 
+	// Determine watch timeout
+	deadline, _ := ctx.Deadline()
```

Comment:
It's possible that no deadline is set here, right? We should be explicit about what happens in that case, either by checking `ok` (`deadline, ok := ctx.Deadline()` ... `if !ok { ... }`) or by adding a comment explaining the behavior here.

Reply:
add a comment here, and change a little bit.